docs/features/wired-mesh.rst

```diff
@@ -56,32 +56,63 @@ the ``mesh`` role to the ``interfaces.*.default_roles`` options in site.conf.
 Commandline
 ===========
 
+Since version 2022.1, the wired network configuration is rebuilt from ``/etc/config/gluon`` on each ``gluon-reconfigure``, and therefore overwritten at least on every firmwareupgrade.
+Every interface has a list of roles assigned to it.
+There are currently three roles available: ``client``, ``mesh`` and ``uplink``.
+
+Be careful: When the client role is assigned to an interface at the same time as other roles (like 'client', 'mesh' in the Mesh-on-LAN example below), the other roles take precedence (enabling mesh, but not client in said example).
+
+The setup/config-mode interface is every interface with the role ``client``, which makes removing it from interfaces not only unnecessary, but generally unrecommended.
+
+In order to make persistent changes to the routers configuration it's necessary to:
+
+* change the sections in ``/etc/config/gluon`` e.g. using uci (see examples below)
+* call ``gluon-reconfigure`` to regenerate ``/etc/config/network`` afterwards
+* apply the networking changes, either through executing ``service network restart`` or by performing a ``reboot`` of the device
+
 Enable Mesh-on-WAN::
 
-  uci set network.mesh_wan.disabled=0
-  uci commit network
+  uci add_list gluon.iface_wan.role='mesh'
+  uci commit gluon
 
 Disable Mesh-on-WAN::
 
-  uci set network.mesh_wan.disabled=1
-  uci commit network
+  uci del_list gluon.iface_wan.role='mesh'
+  uci commit gluon
 
 Enable Mesh-on-LAN::
 
-  uci set network.mesh_lan.disabled=0
-  for ifname in $(cat /lib/gluon/core/sysconfig/lan_ifname); do
-    uci del_list network.client.ifname=$ifname
-  done
-  uci commit network
+  uci add_list gluon.iface_lan.role='mesh'
+  uci commit gluon
 
 Disable Mesh-on-LAN::
 
-  uci set network.mesh_lan.disabled=1
-  for ifname in $(cat /lib/gluon/core/sysconfig/lan_ifname); do
-    uci add_list network.client.ifname=$ifname
-  done
-  uci commit network
+  uci del_list gluon.iface_lan.role='mesh'
+  uci commit gluon
+
+Enable Mesh-on-Single::
+
+  uci add_list gluon.iface_single.role='mesh'
+  uci commit gluon
+
+Disable Mesh-on-Single::
+
+  uci del_list gluon.iface_single.role='mesh'
+  uci commit gluon
+
+Furthermore it is possible to make use of VLANs.
+The following statements would create a VLAN with id 8 on ``eth0`` and emit the mesh network on it::
+
+  uci set gluon.iface_lan_vlan8=interface
+  uci set gluon.iface_lan_vlan8.name='eth0.8'
+  uci add_list gluon.iface_lan_vlan8.role='mesh'
+  uci commit gluon
+
+Other VLAN-interfaces could be configured on the same parent interface in order to have all three roles avaliable on ``eth0`` without having them interfere with each other.
+This feature comes in especially handy for the persistent configuration of virtual machines as offloader for bigger installations.
+
+It is not enough to ``reboot`` the device, calling ``gluon-reconfigure`` before is mandatory in order for the changes to take effect.
```

Comment:
It's a bit redundant, but maybe it helps people, because it is quite important to know, idk

Reply:
I'm not sure about it either.
I felt like it wasn't stressed enough before.